### PR TITLE
DEVX-2465 Add app-to-app tutorial

### PR DESCRIPTION
**Pull request by igorwojda**
Jira https://nexmoinc.atlassian.net/browse/DEVX-2465

## Description

- Add app-to-app tutorial (based on app-to-phone)
- Updates to app-to-phone tutorial and messaging-chat tutorial (mostly rearrange and unify content to share across tutorials)

## Deploy Notes

https://nexmo-develo-devx-2959--5krfmo.herokuapp.com/client-sdk/tutorials/app-to-app/introduction/kotlin

### CODE REVIEW

File: config/tutorials/en/app-to-phone.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-title: Making an in-app voice call
-description: You make a voice call from a web app to a phone.
+title: Making in-app voice call
+description: You make a phone call from a app to another app
```
Comment (tbedford) [inline config/tutorials/en/app-to-phone.yml:2]: Description change looks wrong. I really don't understand why the app-to-phone.yml file should be changed to include info on app to app?

File: _tutorials/en/client-sdk/android-shared/configure-jwt-alice-and-bob/kotlin.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Configure JWTs
+title: Configure JWT
```
Comment (tbedford) [inline _tutorials/en/client-sdk/android-shared/configure-jwt-alice-and-bob/kotlin.md:2]: Not sure why you made this change?

File: _tutorials/en/client-sdk/android-shared/create-ncco.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
+title: Create a NCCO
+description: In this step, you modify your NCCO using GitHub Gist.
+---
+
+# Create a NCCO
+
+A Nexmo Call Control Object (NCCO) is a JSON array that you use to control the flow of a Voice API call. More information on NCCO can be found [here](https://developer.nexmo.com/voice/voice-api/ncco-reference).
```
Comment (tbedford) [inline _tutorials/en/client-sdk/android-shared/create-ncco.md:8]: "https://developer.nexmo.com/voice/voice-api/ncco-reference" -> /voice/voice-api/ncco-reference

File: _tutorials/en/client-sdk/android-shared/create-ncco.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (tbedford) [inline _tutorials/en/client-sdk/android-shared/create-ncco.md:1]: How is the different to the generic 'create NCCO topic' - are there Android specifics?

File: _tutorials/en/client-sdk/android-shared/create-ncco.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            {
+                "type": "phone",
+                "number": "PHONE_NUMBER"
+            }
+        ]
+    }
+]
+```
+
+4) Replace `PHONE_NUMBER` with your phone number (must be in [E.164](https://developer.nexmo.com/concepts/guides/glossary#e-164-format) format, for example: 14155550100 or 447700900001)
```
Comment (tbedford) [inline _tutorials/en/client-sdk/android-shared/create-ncco.md:36]: "https://developer.nexmo.com/concepts/guides/glossary#e-164-format" -> /concepts/guides/glossary#e-164-format

File: _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+---
+title: Make a call
+description: In this step you will make an app call
```
Comment (tbedford) [inline _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md:3]: "In this step you will make an app call" -> In this step you learn how to make an in-app call. 

You could also change 'in-app call' to app-to-app call if that's more accurate.

File: _tutorials/en/client-sdk/app-to-app/recieve-call/kotlin.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+---
+title: Receive a call
+description: In this step you will recieve make an app call
```
Comment (tbedford) [inline _tutorials/en/client-sdk/app-to-app/recieve-call/kotlin.md:3]: "In this step you will recieve make an app call". Receive is spelt incorrectly. Also I would reword: In this step you learn how to receive an in-app call.

File: _tutorials/en/client-sdk/app-to-app/starter-project/kotlin.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+---
+title: The Starter Project
+description: In this step you will clone the starter project
+---
+
+# The Starter Project
+
+To make things easier, a `Starter` project is provided for you. It is a simple Android Studio project that contains an application with the following two screens:
+
+1. Clone this [GitHub project](2. https://github.com/nexmo-community/client-sdk-android-tutorial-voice-app-to-app).
```
Comment (tbedford) [inline _tutorials/en/client-sdk/app-to-app/starter-project/kotlin.md:10]: > `[GitHub project](2. https://github.com/nexmo-community/client-sdk-android-tutorial-voice-app-to-app).`

The link looks wrong. the `2.` appears to have crept in by mistake.

File: _tutorials/en/client-sdk/app-to-phone/make-call/kotlin.md [hunk 9]
```diff
@@ -1,9 +1,9 @@
 ---
 title: Make a call
-description: In this step you make a phone call
+description: In this step you will make a phone call
```
Comment (tbedford) [inline _tutorials/en/client-sdk/app-to-phone/make-call/kotlin.md:3]: "In this step you will make a phone call". Reword to "In this step you learn how to make a phone call"

File: _tutorials/en/client-sdk/app-to-phone/starter-project/kotlin.md [hunk 10]
```diff
@@ -7,8 +7,8 @@ description: In this step you will clone the starter project
 
 To make things easier, a `Starter` project is provided for you. It is a simple Android Studio project that contains an application with the following two screens:
 
-1. Clone this [GitHub project](https://github.com/nexmo-community/client-sdk-android-tutorial-voice).
-2. Open the project in the `Android Studio` - navigate to menu `File -> Open` and select the `kotlin-start` folder from cloned repository.
+1. Clone this [GitHub project](2. https://github.com/nexmo-community/client-sdk-android-tutorial-voice-app-to-phone).
```
Comment (tbedford) [inline _tutorials/en/client-sdk/app-to-phone/starter-project/kotlin.md:10]: Link has errant number 2 in it?

File: config/tutorials/en/in-app-messaging/kotlin.yml [hunk 11]
```diff
@@ -43,9 +43,11 @@ conclusion:
 
     # What's next?
 
-    For the complete version please see the Android Studio project in the `kotlin-completed` folder of the `git` repository you've cloned ([GitHub](https://github.com/nexmo-community/client-sdk-android-tutorial-messaging)).
+    For the complete version please see the Android Studio project in the `kotlin-completed` folder of 
+    the [git repository](https://github.com/nexmo-community/client-sdk-android-tutorial-messaging) you've cloned.
 
-    You can do a lot more with the Client SDK. See [Client SDK documentation](/client-sdk/overview).
+    You can do a lot more with the Client SDK. See [Client SDK documentation](/client-sdk/overview) or review
+    [Android Client SDK documentation](https://developer.nexmo.com/sdk/stitch/android/index).
```
Comment (tbedford) [inline config/tutorials/en/in-app-messaging/kotlin.yml:50]: I think '/sdk/stitch/android/index' should work for the link.

File: _tutorials/en/client-sdk/android-shared/create-ncco.md (hunk 3 above)
Comment (igorwojda) [reply _tutorials/en/client-sdk/android-shared/create-ncco.md:8]: Above file updated. 

We have this link `https://developer.nexmo.com/voice/voice-api/ncco-reference`  in more places. Should I update all?
image

File: _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md (hunk 6 above)
Comment (igorwojda) [reply _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md:3]: Updated and also changed to `app-to-app` 

I am generally fan of
`app-to-app` / `app-to-phone` / `phone-to-app` as they are quite consistent and easy tto get, this `in-app call` makes me confused. We should change all the NDP docs IMO (as separate PR). Do you think it's god idea?

File: config/tutorials/en/in-app-messaging/kotlin.yml (hunk 11 above)
Comment (igorwojda) [reply config/tutorials/en/in-app-messaging/kotlin.yml:50]: Yes works

File: _tutorials/en/client-sdk/android-shared/create-ncco.md (hunk 4 above)
Comment (igorwojda) [reply _tutorials/en/client-sdk/android-shared/create-ncco.md:1]: Good catch - I have unified JS and Android ones

File: _tutorials/en/client-sdk/android-shared/create-ncco.md (hunk 3 above)
Comment (tbedford) [reply _tutorials/en/client-sdk/android-shared/create-ncco.md]: Yes please!

File: _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md (hunk 6 above)
Comment (tbedford) [reply _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md]: Yes, I agree, I tried to always use app-to-app / app-to-phone / phone-to-app as it's much more obvious what's going on.

File: _tutorials/en/client-sdk/android-shared/create-ncco.md (hunk 3 above)
Comment (igorwojda) [reply _tutorials/en/client-sdk/android-shared/create-ncco.md]: Done 
https://nexmoinc.atlassian.net/browse/DEVX-3434

File: _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md (hunk 6 above)
Comment (igorwojda) [reply _tutorials/en/client-sdk/app-to-app/make-call/kotlin.md]: Ticket created
https://nexmoinc.atlassian.net/browse/DEVX-3435

File: _tutorials/en/client-sdk/app-to-phone/create-ncco.md [hunk 12]
```diff
@@ -5,9 +5,9 @@ description: In this step, you modify your NCCO using GitHub Gist.
 
 # Create a NCCO
```
Comment (tbedford) [inline _tutorials/en/client-sdk/app-to-phone/create-ncco.md:6]: The number goes out on this topic. 1, 2, 3, 1, 2, 6, 7. Also the last screenshot looks a bit too "in your face" - perhaps we could reduce the size a little?

File: _tutorials/en/client-sdk/app-to-phone/create-ncco.md [hunk 13]
```diff
@@ -5,9 +5,9 @@ description: In this step, you modify your NCCO using GitHub Gist.
...
 
-A Nexmo Call Control Object (NCCO) is a JSON array that you use to control the flow of a Voice API call. More information on NCCO can be found [here](https://developer.nexmo.com/voice/voice-api/ncco-reference).
+A Nexmo Call Control Object (NCCO) is a JSON array that you use to control the flow of a Voice API call. More information on NCCO can be found [here](/voice/voice-api/ncco-reference).
 
-The NCCO must be public and accessible by the internet. To accomplish that, we will be using [GitHub Gist](https://gist.github.com/):
+The NCCO must be public and accessible by the internet. To accomplish that, we will be using [GitHub Gist](https://gist.github.com/) that provides easy way to host the configuration:
```
Comment (tbedford) [inline _tutorials/en/client-sdk/app-to-phone/create-ncco.md:10]: * we will be using -> you will be using. 
* "that provides easy way" -> provides a convenient way. (sounds picky, but we generally don't make judgements on what's easy or difficult).

File: _tutorials/en/client-sdk/app-to-phone/create-ncco.md (hunk 12 above)
Comment (tbedford) [reply _tutorials/en/client-sdk/app-to-phone/create-ncco.md:6]: Checking the review app, the numbering still looks out I think.
